server/routerlicious/packages/services-utils/src/auth.ts

```diff
@@ -143,6 +148,81 @@ function getTokenFromRequest(request: Request): string {
 	return tokenMatch[1];
 }
 
+const defaultMaxTokenLifetimeSec = 60 * 60; // 1 hour
+
+export async function verifyToken(
+	tenantId: string,
+	documentId: string,
+	token: string,
+	tenantManager: ITenantManager,
+	options: IVerifyTokenOptions,
+): Promise<void> {
+	if (options.requireDocumentId && !documentId) {
+		throw new NetworkError(403, "Missing documentId.");
+	}
+
+	let tokenLifetimeMs: number | undefined;
+	const logProperties = getLumberBaseProperties(documentId, tenantId);
+	try {
+		const claims = validateTokenClaims(token, documentId, tenantId, options.requireDocumentId);
+		if (options.requireTokenExpiryCheck) {
+			let maxTokenLifetimeSec = options.maxTokenLifetimeSec;
+			if (!maxTokenLifetimeSec) {
+				Lumberjack.error(
+					`Missing/Invalid maxTokenLifetimeSec=${maxTokenLifetimeSec} in options. Set to default=${defaultMaxTokenLifetimeSec}`,
+					logProperties,
+				);
+				maxTokenLifetimeSec = defaultMaxTokenLifetimeSec;
+			}
+			tokenLifetimeMs = validateTokenClaimsExpiration(claims, maxTokenLifetimeSec);
+		}
+
+		// Check token cache first
+		if ((options.enableTokenCache || options.ensureSingleUseToken) && options.tokenCache) {
+			const cachedToken = await options.tokenCache.get(token).catch((error) => {
+				Lumberjack.error("Unable to retrieve cached JWT", logProperties, error);
+				return false;
+			});
+
+			if (cachedToken) {
+				Lumberjack.info("Token cache hit", logProperties);
+				if (options.ensureSingleUseToken) {
+					throw new NetworkError(403, "Access token has already been used.");
+				}
+				return;
+			}
+		}
+
+		await tenantManager.verifyToken(claims.tenantId, token);
+
+		// Update token cache
+		if ((options.enableTokenCache || options.ensureSingleUseToken) && options.tokenCache) {
+			Lumberjack.info("Token cache miss", logProperties);
+			const tokenCacheKey = token;
+			options.tokenCache
+				.set(
+					tokenCacheKey,
+					"used",
+					tokenLifetimeMs !== undefined ? Math.floor(tokenLifetimeMs / 1000) : undefined,
```

Comment:
nit: If the time to cache a token value turns out to be undefined, then use some default value.

Reply:
We are using Redis cache here and it has some logic to set the default timeout for its entry.